src/main/java/randoop/sequence/Sequence.java

```diff
@@ -699,18 +699,64 @@ private Statement getStatementWithInputs(int index) {
    */
   public List<Variable> allVariablesForTypeLastStatement(Type type, boolean onlyReceivers) {
     List<Variable> possibleVars = new ArrayList<>(this.lastStatementVariables.size());
-    for (Variable i : this.lastStatementVariables) {
-      Statement s = statements.get(i.index);
-      Type outputType = s.getOutputType();
-      if (type.isAssignableFrom(outputType)
-          && !(onlyReceivers && outputType.isNonreceiverType())
-          && !(onlyReceivers && getCreatingStatement(i).isNonreceivingInitialization())) {
-        possibleVars.add(i);
+    for (Variable var : this.lastStatementVariables) {
+      if (matchesVariable(var, type, onlyReceivers)) {
+        possibleVars.add(var);
       }
     }
     return possibleVars;
   }
 
+  /**
+   * Return the first value of type {@code type} that is produced by, or might be side-effected by,
```

Comment:
As noted elsewhere, I think the logic that is needed is to return the single value produced by the method call in the last statement (which should be to a non-void method).